### PR TITLE
[GST-2192] Updates make and test

### PR DESCRIPTION
**Pull request by lobsterdore**
Adjusts makefile and vagrant tests to match latest sansible
layout.

Also added Xenial to list of tested distros and updated readme.

### CODE REVIEW

**Comment by quater** [comment]
`make test` failed with below error:

```
TASK: [sansible.gocd_server | Install go-server] ****************************** 

failed: [default] => {"failed": true}
stderr: W: --force-yes is deprecated, use one of the options starting with --allow instead.
E: Failed to fetch https://download.go.cd/binaries/16.1.0-2855/deb/go-server-16.1.0-2855.deb  transfer closed with 119295513 bytes remaining to read

E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?

stdout: Reading package lists...
Building dependency tree...
Reading state information...
Suggested packages:
  go-agent
The following NEW packages will be installed:
  go-server
0 upgraded, 1 newly installed, 0 to remove and 24 not upgraded.
Need to get 141 MB of archives.
After this operation, 419 MB of additional disk space will be used.
Get:1 https://download.go.cd  go-server 16.1.0-2855 [141 MB]
Err:1 https://download.go.cd  go-server 16.1.0-2855
  transfer closed with 119295513 bytes remaining to read

msg: '/usr/bin/apt-get -y -o "Dpkg::Options::=--force-confdef" -o "Dpkg::Options::=--force-confold" --force-yes  install 'go-server=16.1.*'' failed: W: --force-yes is deprecated, use one of the options starting with --allow instead.
E: Failed to fetch https://download.go.cd/binaries/16.1.0-2855/deb/go-server-16.1.0-2855.deb  transfer closed with 119295513 bytes remaining to read

E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?


FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/home/hkuehn/test.retry

default                    : ok=17   changed=13   unreachable=0    failed=1   

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
Makefile:25: recipe for target 'vagrant' failed
make: *** [vagrant] Error 1
```

Ubuntu 16.04 requires more fine grained `apt` options. Alternative options are `--allow-downgrades`,`--allow-remove-essential` ,`--allow-change-held-packages`.

http://manpages.ubuntu.com/manpages/xenial/man8/apt-get.8.html

**Comment by lobsterdore** [comment]
Hmmm I just tried this a few times with the bento 16 box and I didn't get this error, can you re-test when you get a chance?

File: tests/vagrant/Vagrantfile [hunk 1]
```diff
@@ -35,17 +21,15 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
...
 		ansible.verbose = "#{ENV['VERBOSE']}" if ENV['VERBOSE']
 	end
 
 	# Test "configure" tag
 	config.vm.provision "ansible" do |ansible|
-		ansible.playbook = "test.yml"
-		ansible.tags = [ "configure" ]
-		ansible.extra_vars = local_vars
+		ansible.playbook = "../test.yml"
+		ansible.tags = [ "configure", "asert" ]
```
Comment (quater) [inline tests/vagrant/Vagrantfile]: This is a typo, it needs to be `assert`.

Below is the error I am getting when running `make test`.

```
==> default: Running provisioner: ansible...
    default: Running ansible-playbook...
ERROR: tag(s) not found in playbook: asert.  possible values: assert,build,maintain
Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
Makefile:25: recipe for target 'vagrant' failed
make: *** [vagrant] Error 1
```

**Comment by quater** [comment]
I have run the test again and did not encounter the above error but instead got the error

```
==> default: Running provisioner: ansible...
    default: Running ansible-playbook...
ERROR: tag(s) not found in playbook: asert.  possible values: assert,build,maintain
Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
Makefile:25: recipe for target 'vagrant' failed
make: *** [vagrant] Error 1
```

Please see inline comment for details.

File: tests/test.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      apt:
+        cache_valid_time: 1800
+        update_cache: yes
+      tags:
+        - build
+
+  roles:
+    - role: sansible.gocd_server
+
+  post_tasks:
```
Comment (quater) [inline tests/test.yml:18]: This breaks on Ubuntu 16.04 due to systemd being used. I guess you can do the same as you implemented with https://github.com/sansible/gocd_agent/blob/5928033b1ffac2f892b70b6d642646f998be1072/tests/test.yml

**Comment by quater** [comment]
`make test` failed with below error:

```
RUNNING HANDLER [sansible.gocd_server : restart go-server] *********************
fatal: [default]: FAILED! => {"changed": false, "failed": true, "msg": "Job for go-server.service failed because the control process exited with error code. See \"systemctl status go-server.service\" and \"journalctl -xe\" for details.\n"}
```

The `systemctl status go-server.service` output was:

```
root@vagrant:~# systemctl status go-server.service
● go-server.service - LSB: Go Server
   Loaded: loaded (/etc/init.d/go-server; bad; vendor preset: enabled)
   Active: failed (Result: exit-code) since Mon 2016-10-17 17:35:25 UTC; 1min 20s ago
     Docs: man:systemd-sysv-generator(8)
  Process: 6419 ExecStop=/etc/init.d/go-server stop (code=exited, status=0/SUCCESS)
  Process: 6440 ExecStart=/etc/init.d/go-server start (code=exited, status=255)

Oct 17 17:34:55 vagrant systemd[1]: Starting LSB: Go Server...
Oct 17 17:34:55 vagrant su[6445]: Successful su for go by root
Oct 17 17:34:55 vagrant su[6445]: + ??? root:go
Oct 17 17:34:55 vagrant su[6445]: pam_unix(su:session): session opened for user go by (uid=0)
Oct 17 17:34:55 vagrant go-server[6440]: using default settings from /etc/default/go-server
Oct 17 17:35:25 vagrant go-server[6440]: Error starting Go Server.
Oct 17 17:35:25 vagrant systemd[1]: go-server.service: Control process exited, code=exited status=255
Oct 17 17:35:25 vagrant systemd[1]: Failed to start LSB: Go Server.
Oct 17 17:35:25 vagrant systemd[1]: go-server.service: Unit entered failed state.
Oct 17 17:35:25 vagrant systemd[1]: go-server.service: Failed with result 'exit-code'.
```

Application log:

`less /var/log/go-server/go-server.log`

```
2016-10-17 17:34:50,922 ERROR [Thread-70] PluginSettingsMetadataLoader:72 - Failed to fetch Plugin Settings metadata for plugin : yum
java.lang.RuntimeException: Interaction with plugin with id 'yum' implementing 'package-repository' extension failed while requesting for 'go.plugin-settings.get-configuration'. Reason: [The plugin sent a response that could not be understood by Go. Plugin returned with code '400' and the following response: 'Invalid request name go.plugin-settings.get-configuration']
        at com.thoughtworks.go.plugin.access.PluginRequestHelper.submitRequest(PluginRequestHelper.java:38)
        at com.thoughtworks.go.plugin.access.common.settings.AbstractExtension.getPluginSettingsConfiguration(AbstractExtension.java:45)
        at com.thoughtworks.go.plugin.access.packagematerial.PackageAsRepositoryExtension.getPluginSettingsConfiguration(PackageAsRepositoryExtension.java:51)
        at com.thoughtworks.go.plugin.access.common.settings.PluginSettingsMetadataLoader.fetchPluginSettingsMetaData(PluginSettingsMetadataLoader.java:63)
        at com.thoughtworks.go.plugin.access.common.settings.PluginSettingsMetadataLoader.pluginLoaded(PluginSettingsMetadataLoader.java:47)
        at com.thoughtworks.go.plugin.infra.DefaultPluginManager$FilterChangeListener.pluginLoaded(DefaultPluginManager.java:262)
        at com.thoughtworks.go.plugin.infra.FelixGoPluginOSGiFramework$2.execute(FelixGoPluginOSGiFramework.java:353)
        at org.apache.commons.collections.CollectionUtils.forAllDo(CollectionUtils.java:389)
        at com.thoughtworks.go.plugin.infra.FelixGoPluginOSGiFramework.getBundle(FelixGoPluginOSGiFramework.java:111)
        at com.thoughtworks.go.plugin.infra.FelixGoPluginOSGiFramework.loadPlugin(FelixGoPluginOSGiFramework.java:97)
```

File: README.md [hunk 3]
```diff
@@ -179,20 +183,24 @@ create ~/.aws/credentials for you. But we strongly advise to use
...
-If you want to work on this role, please start with running
-`make watch`. This will watch for any file changes and re-provision
-vagrant box.
+If you want to work on this role you will find several test targets 
+in the Makfefile.
 
-If you want to test the AWS backup/restore script locally you need
-to create a file `/tests/vagrant/local_vars.yml` with
+`make test` - This will bring up the Vagrant machine and run a test
+at the end
```
Comment (quater) [inline README.md]: To make it more readable, two trailing spaces are required for a line break.

**Comment by quater** [comment]
With the current most recent code line https://github.com/sansible/gocd_server/pull/20/commits/de4d82583c4e1d62c633959899df5f36e5e6c2a7 I run the `UBUNTU_VERSION=16 make test` and it completed without errors.

Only a minor change required in the README.md file to make it more readable and we are good to go.

**Comment by quater** [comment]
LGTM :+1:

**Pull request merged**